src/gui/sdlmain.cpp

```diff
@@ -3126,47 +3170,46 @@ DosBox::Rect GFX_CalcViewportInPixels(const int canvas_width_px,
 	assert(std::isfinite(draw_scale_y));
 
 	// Limit the window to the user's desired viewport, if configured
-	const auto restricted_dims_px =
-	        restrict_to_viewport_resolution(canvas_width_px, canvas_height_px);
-
-	const auto bounds_w_px = restricted_dims_px.x;
-	const auto bounds_h_px = restricted_dims_px.y;
+	const auto bounds_px = calc_restricted_viewport_size_in_pixels(canvas_size_px);
```

Comment:
this is a nice cleanup! finally collapsing all these free-floating dimensions into structs. :1st_place_medal: